mvpa2/base/dataset.py

```diff
@@ -207,8 +211,17 @@ def __init__(self, samples, sa=None, fa=None, a=None):
 
         """
         # conversions
-        if isinstance(samples, list):
-            samples = np.array(samples)
+        if isinstance(samples, list) or isinstance(samples, tuple):
+            if np.all([is_datasetlike(e) for e in samples]):
+                # special case dataset of datasets
+                temp = np.empty((len(samples),), dtype=np.object)
+                for i, s in enumerate(samples):
+                    temp[i] = s
+                samples = temp
+                del temp
```

Comment:
is it really needed here?  isn't that dataset contains only references to the original ones, and temp will simply a lightweight beast so ok to be picked up by gc at any convenience?